content/blog/2020-03-06-wasm/index.md

```diff
@@ -0,0 +1,207 @@
+could do these feats from the convenience of the Chrome browser? Well, 
+then you might have just heard that now you can do it in (almost) any browser out there. 
+For me, that's what WebAssembly brings.
+
+In this article, I don't expect to talk much about the language (or binary code format) 
+itself, but implications of its compilation which makes 
+things like [Google Earth][earth] on a browser possible.
+
+<img src="google-earth.png" width="700" >
+
+[earth]: https://www.google.com/earth/
+
+## What is WebAssembly?
+WebAssembly is a binary code format to transfer web applications from the 
+server to the browser. It is incorporated in modern browsers to be used in 
+tandem with existing JavaScript applications, and uses components from 
+existing JavaScript engines to interpret and execute. It has taken the world 
+wide web by storm as
+- it is rolled out by four major browsers (Chrome, Edge, Mozilla and Safari)
+i.e. platform independent
+- it is programming model independent
+- it is hardware independent
+
+What this means is you could write a web application in C, compile it to 
+WebAssembly and use it on any browser on any hardware. This gives much more 
+performance in general than a JavaScript program, which has type 
+classification overheads, performance implications based on which browser 
+you target and how performant is your interpreter.
+
+## Is WebAssembly just C on browsers?
+WebAssembly does brings the performance of pre-compiled languages to web browsers. 
+But it does so side-by-side with traditional Javascript running on the same engines.
+This allows browsers to have best of both worlds, drive the Ferrari on the race track- 
+run performant WebAssembly when it needs performance, or drive the Prius to the 
+grocery store- quickly getting a simple web application up and running.
+
+With WebAssembly, you could use JavaScript for fast development, but also 
+use C where you need performance. You could use C where static typing is
+useful, but JavaScript where dynamic typing is a necessacity for productivity.
+And you could combine these modules, to balance productivity and performance,
+as both these styles can now be executed in the same compiler flow.
+So WebAssembly is more than just C on browsers, it's a carefully coordinated, 
+well-designed, massive engineering effort to reconcile two worlds at odds.
+
+## How does it differ from existing tools?
+So WebAssembly is portable, but doesn't JVM(Java virtual machine) already 
+offer portability? Even though the overall goal has some similarities, Java 
+cannot be run on a browser without plugins and has doesn't have the language
+flexibilty WebAssembly provides. Moreover, validation and memory safety makes 
+WebAssembly more secure from vulnerabilities. Structured control flow and 
+single pass compilation and validation places WebAssembly as a mature option 
+compared to existing bytecode formats such as JVM and CIL (Common Intermediate Language).
```

Comment:
Again, a little more substantiation would go a long way here—what exactly is it about single-pass compilation/validation and structured control flow that make was "better"?